### PR TITLE
ACTIN-1248: PR - renamed approvalStatus to evidenceLevelDetails and a…

### PR DESCRIPTION
**Pull request by spdeleeuw**
…dded data structuring

Post-merge changes to ACTIN-1248 as suggested in https://github.com/hartwigmedical/actin/pull/735/files and in-person by @kduyvesteyn 

- Renamed approvalStatus to evidenceLevelDetails for clarity
- Data structuring on SERVE side to easily handle different types of approval statuses downstream in ACTIN

### CODE REVIEW

**Review by pauldwolfe — APPROVED** [review]

**Comment by spdeleeuw** [comment]
@kduyvesteyn FYI the tables (column names + transformed data) look good to me when testing on serve vm

**Review by kzuberihmf — APPROVED** [review]

File: algo/src/main/java/com/hartwig/serve/sources/ckb/ActionableEvidenceFactory.java [hunk 1]
```diff
@@ -202,15 +202,34 @@ static EvidenceLevel resolveLevel(@Nullable String evidenceLabel) {
 
     @Nullable
     @VisibleForTesting
-    static ApprovalStatus resolveApprovalStatus(@Nullable String approvalStatusLabel) {
-        if (approvalStatusLabel == null) {
+    static EvidenceLevelDetails resolveEvidenceLevelDetails(@Nullable String evidenceLevelDetails) {
```
Comment (kduyvesteyn) [inline algo/src/main/java/com/hartwig/serve/sources/ckb/ActionableEvidenceFactory.java:205]: You feed in the approval status here, would suggest to rename input variable

File: algo/src/main/java/com/hartwig/serve/sources/ckb/ActionableEvidenceFactory.java [hunk 2]
```diff
@@ -202,15 +202,34 @@ static EvidenceLevel resolveLevel(@Nullable String evidenceLabel) {
...
+            return EvidenceLevelDetails.CLINICAL_STUDY.evidenceLevelDetailsCreator(evidenceLevelDetails);
+        }
+        else if (evidenceLevelDetails.contains("guideline")) {
+            return EvidenceLevelDetails.GUIDELINE.evidenceLevelDetailsCreator(evidenceLevelDetails);
+        }
+        else if (evidenceLevelDetails.contains("fda approved")) {
+            return EvidenceLevelDetails.FDA_APPROVED.evidenceLevelDetailsCreator(evidenceLevelDetails);
+        }
+        else if (evidenceLevelDetails.contains("fda contraindicated")) {
+            return EvidenceLevelDetails.FDA_CONTRAINDICATED.evidenceLevelDetailsCreator(evidenceLevelDetails);
```
Comment (kduyvesteyn) [inline algo/src/main/java/com/hartwig/serve/sources/ckb/ActionableEvidenceFactory.java:227]: This is incorrect and dodgy. You should not overwrite properties of an enum that should be private.

If you need to retain the source approval status you should add that as a separate field.

File: datamodel/src/main/java/com/hartwig/serve/datamodel/EvidenceLevelDetails.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public EvidenceLevelDetails evidenceLevelDetailsCreator(String description) {
+        this.description = description;
+        return this;
+    }
```
Comment (kduyvesteyn) [inline datamodel/src/main/java/com/hartwig/serve/datamodel/EvidenceLevelDetails.java:25]: The description should be a private final variable (annotated with NotNull), and not settable. See also other comment

**Pull request merged**